packages/database/src/core/Repo.ts

```diff
@@ -470,32 +476,42 @@ export function repoGetValue(repo: Repo, query: QueryContext): Promise<Node> {
       const node = nodeFromJSON(payload).withIndex(
         query._queryParams.getIndex()
       );
-      // if this is a filtered query, then overwrite at path
-      if (query._queryParams.loadsAllData()) {
-        syncTreeApplyServerOverwrite(repo.serverSyncTree_, query._path, node);
-      } else {
-        // Simulate `syncTreeAddEventRegistration` without events/listener setup.
-        // We do this (along with the syncTreeRemoveEventRegistration` below) so that
-        // `repoGetValue` results have the same cache effects as initial listener(s)
-        // updates.
-        const tag = syncTreeRegisterQuery(repo.serverSyncTree_, query);
-        syncTreeApplyTaggedQueryOverwrite(
-          repo.serverSyncTree_,
-          query._path,
-          node,
-          tag
-        );
-        // Call `syncTreeRemoveEventRegistration` with a null event registration, since there is none.
-        // Note: The below code essentially unregisters the query and cleans up any views/syncpoints temporarily created above.
-      }
-      const cancels = syncTreeRemoveEventRegistration(
+      /**
+       * Below we simulate the actions of an `onlyOnce` `onValue()` event where:
+       * we add an event registration,
+       * update data at the path,
+       * trigger the events,
+       * and then cleanup the SyncTree
+       */
+      const events = syncTreeAddGetRegistration(
+        repo,
+        query,
+        node,
+        eventRegistration
+      );
+      /*
+       * We need to raise events in the scenario where `get()` is called at a parent path, and
+       * while the `get()` is pending, `onValue` is called at a child location. While get() is waiting
+       * for the data, `onValue` will register a new event. Then, get() will come back, and update the syncTree
+       * and its corresponding serverCache, including the child location where `onValue` is called. Then,
+       * `onValue` will receive the event from the server, but look at the syncTree and see that the data received
+       * from the server is already at the SyncPoint, and so the `onValue` callback will never get fired.
+       * Calling `eventQueueRaiseEventsForChangedPath()` is the correct way to propagate the events and
+       * ensure the corresponding child events will get fired.
+       */
+      eventQueueRaiseEventsForChangedPath(
+        repo.eventQueue_,
+        query._path,
+        events
+      );
+      syncTreeRemoveEventRegistration(
+        // syncTreeAddGetRegistration calls syncTreeAddEventRegistration. We need to clean that up here.
```

Comment:
I think we should inline syncTreeAddGetRegistration, then we can remove this comment.